components/omega/src/base/Halo.h

```diff
@@ -0,0 +1,330 @@
+#ifndef OMEGA_HALO_H
+#define OMEGA_HALO_H
+//===-- base/Halo.h - halo exchange definitions -----------------*- C++ -*-===//
+//
+/// \file
+/// \brief Defines classes and methods for performing halo exchanges
+///
+/// This header defines classes and methods needed for exchanging the halo
+/// elements of every supported array type in OMEGA with elements of the array
+/// on neighboring mesh partitions. The Halo class constructor creates a Halo
+/// object based on a given machine environment and parallel decomposition.
+/// The Halo object contains all the info needed for performing a halo
+/// exchange The halo exchanges are carried out via non-blocking MPI library
+/// routines. The Halo class public member function exchangeFullArrayHalo
+/// which is called by the user to perform halo exchanges is a template
+/// function and thus is fully defined in this header.
+///
+//
+//===----------------------------------------------------------------------===//
+
+#include "DataTypes.h"
+#include "Decomp.h"
+#include "Logging.h"
+#include "MachEnv.h"
+#include "mpi.h"
+
+namespace OMEGA {
+
+// Set the default MPI real data type as single or double precision based on
+// the default real data type used by OMEGA.
+#ifdef SINGLE_PRECISION
+constexpr MPI_Datatype MPI_RealKind = MPI_FLOAT;
+#else
+constexpr MPI_Datatype MPI_RealKind = MPI_DOUBLE;
```

Comment:
On Chrysalis, the Intel mpi implementation apparently doesn't define the MPI type constants in a way that allows compile-time evaluation so generated an error here. I guess we need to use "static const" is place of constexpr here. Not the first time we've hit issues with Intel MPI.